docs(PULL_REQUEST_TEMPLATE): add Topic changes and Effects for driving behavior (#3423)
autowarefoundation/autoware: .github/PULL_REQUEST_TEMPLATE/small-change.md

Not applicable.

> ## Effects on driving behavior

Reviewer: I'm not sure if it's limited to `driving`. I personally feel `system` is safer.

```suggestion
## Effects on system behavior
```

Author: Looks better!

> ## Effects for driving behavior

Author: ```suggestion
## Effects on driving behavior
```

> <!-- Describe how this PR affects the outputs of driving behavior. -->

Reviewer: Related to https://github.com/autowarefoundation/autoware/pull/3423/files#r1164888354.

```suggestion
<!-- Describe how this PR affects the system behavior. -->
```

> No effects.

Reviewer: I think it sometimes has a little effect, so rephrasing it to something might be better.

```suggestion
Not applicable.
```
